app/routes.py

```diff
@@ -1,2 +1,266 @@
-from flask import Blueprint
+from app import db 
+from flask import Blueprint, request, make_response, jsonify
+from app.models.task import Task 
+from app.models.goal import Goal
+from datetime import datetime
+import os 
+import requests
 
+
+
+task_list_bp = Blueprint("Task",__name__)
+
+@task_list_bp.route("/tasks", methods=["POST"])
+def create_a_task(): 
+    request_body = request.get_json()
+
+    if not ("title" in request_body
+            and "description" in request_body
+            and "completed_at" in request_body):
+    
+       
+         return make_response(jsonify({
+            "details": "Invalid data"
+        }), 400)
+        
+
+    new_task = Task.from_json(request_body)
+    # new_task = Task(title=request_body["title"],
+    #             description=request_body["description"],
+    #             completed_at=request_body["completed_at"])
+
+    db.session.add(new_task)
+    db.session.commit()
+
+  
+
+    response = {
+            "task": new_task.as_json()
+            
+               }
+    return make_response(jsonify(response), 201)
+
+
+
+
+@task_list_bp.route("/tasks", methods=["GET"])
+def retrieve_all_tasks(): 
+    if "filter" in request.args: 
+        # tasks = Task.query.all()
+        # [task for task in tasks]
+        # tasks = [task for task in tasks if task.title == request.args["filter"]]
+        tasks = Task.query.filter_by(title = request.args["filter"]).all() 
+    elif "sort" in request.args:
+        if request.args["sort"] == "desc":
+            tasks = Task.query.order_by(Task.title.desc()).all()
+        elif request.args["sort"] == "id": 
+            tasks = Task.query.order_by(Task.id.asc()).all()
+        else:
+            tasks = Task.query.order_by(Task.title.asc()).all()
+    else:
+        tasks = Task.query.all()
+    
+
+    # return jsonify([
+    #     {
+    #         "id": q.id,
+    #         "title": q.title,
+    #         "description": q.description,
+    #         "is_complete": q.is_complete()
+    #     } for q in tasks
+    # ])
+
+    return jsonify([
+        task.as_json() for task in tasks
+    ])
+
+
+@task_list_bp.route("/tasks/<task_id>", methods=["GET", "PUT", "DELETE"])
```

Comment:
I suggest being consistent with either having a function for each path (with if statements for the function), or (preferably) making a separate function for each path/method combination.